File: LectureCodes/NumQuad/legendre/Matlab/legendre.m
function V= legendre(n,x)
V = ones(size(x)); V = [V; x];
for j=1:n-1
  V = [V; ((2*j+1)/(j+1)).*x.*V(end,:) - j/(j+1)*V(end-1,:)]; end

File: LectureCodes/NumQuad/legendre/Python/legendre.py
import numpy as np


def legendre(n, x):
    v = np.vstack([np.ones(x.shape), x])
    for j in range(1, n):
        v = np.vstack([v, ((2 * j + 1) / (j + 1.0)) * x * v[-1, :] -
                       j / (j + 1.0) * v[-2, :]])
    return v
